playbooks/roles/ambari-blueprint/templates/blueprint_dynamic.j2

```diff
@@ -785,25 +858,83 @@
       }
     },
 {% endif %}
+    {
+      "ams-site" : {
+          "phoenix.spool.directory" : "{{ base_tmp_dir }}"
+      }
+    },
+    {
+      "ams-env" : {
+        "metrics_collector_log_dir" : "{{ base_log_dir }}/ambari-metrics-collector",
+        "metrics_monitor_log_dir" : "{{ base_log_dir }}/ambari-metrics-monitor"
+      }
+    },
+    {
+      "ams-hbase-site" : {
+          "hbase.rootdir" : "file://{{ base_metrics_dir }}/ambari-metrics-collector/hbase",
+          "hbase.tmp.dir" : "{{ base_tmp_dir }}/hbase-tmp"
+      }
+    },
+    {
+      "ams-hbase-env" : {
+        "hbase_log_dir" : "{{ base_log_dir }}/ambari-metrics-collector"
+      }
+    },
+    {
+      "ams-grafana-env" : {
+        "metrics_grafana_log_dir" : "{{ base_log_dir }}/ambari-metrics-grafana",
+        "metrics_grafana_data_dir" : "{{ base_metrics_dir }}/ambari-metrics-grafana"
+      }
+    },
+    {
+      "zookeeper-env" : {
+        "zk_log_dir" : "{{ base_log_dir }}/zookeeper"
+      }
+    },
+    {
+      "kafka-env" : {
+        "kafka_log_dir" : "{{ base_log_dir }}/kafka"
+      }
+    },
     {
       "storm-site": {
 {% if rangeradmin_hosts|length > 0 and ranger_options.enable_plugins|default(no) and (security|lower == "mit-kdc" or security|lower == "active-directory") %}
         "nimbus.authorizer" : "org.apache.ranger.authorization.storm.authorizer.RangerStormAuthorizer",
 {% endif %}
-        "storm.zookeeper.retry.intervalceiling.millis" : "30000"
+        "storm.zookeeper.retry.intervalceiling.millis" : "30000",
+        "storm.local.dir" : "{{ base_data_dir }}/storm"
+      }
+    },
+    {
+      "storm-env" : {
+        "storm_log_dir" : "{{ base_log_dir}}/storm"
       }
     },
     {
       "kafka-broker": {
 {% if rangeradmin_hosts|length > 0 and ranger_options.enable_plugins|default(no) %}
         "authorizer.class.name" : "org.apache.ranger.authorization.kafka.authorizer.RangerKafkaAuthorizer",
 {% endif %}
-        "zookeeper.session.timeout.ms" : "30000"
+        "zookeeper.session.timeout.ms" : "30000",
+        "log.dirs" : "{{ base_data_dir }}/kafka-logs",
+        "dataDir" : "{{ base_data_dir }}/zookeeper"
```

Comment:
is this line needed here? ( looks like an unintentional dupe from Line 926)

Reply:
You're right, i'll correct that, thanks!